script/tool/test/update_excerpts_command_test.dart

```diff
@@ -7,301 +7,87 @@ import 'package:file/file.dart';
 import 'package:file/memory.dart';
 import 'package:flutter_plugin_tools/src/common/core.dart';
 import 'package:flutter_plugin_tools/src/update_excerpts_command.dart';
-import 'package:mockito/mockito.dart';
+import 'package:path/path.dart' as path;
 import 'package:test/test.dart';
 
 import 'common/package_command_test.mocks.dart';
 import 'mocks.dart';
 import 'util.dart';
 
-void main() {
-  late FileSystem fileSystem;
-  late Directory packagesDir;
-  late RecordingProcessRunner processRunner;
-  late CommandRunner<void> runner;
-
-  setUp(() {
-    fileSystem = MemoryFileSystem();
-    packagesDir = createPackagesDirectory(fileSystem: fileSystem);
-    final MockGitDir gitDir = MockGitDir();
-    when(gitDir.path).thenReturn(packagesDir.parent.path);
-    processRunner = RecordingProcessRunner();
-    final UpdateExcerptsCommand command = UpdateExcerptsCommand(
+Future<void> testInjection(String before, String source, String after, { bool failOnChange = false }) async {
+  final FileSystem fileSystem = MemoryFileSystem();
+  final Directory packagesDir = createPackagesDirectory(fileSystem: fileSystem);
+  final CommandRunner<void> runner = CommandRunner<void>('', '')
+    ..addCommand(UpdateExcerptsCommand(
       packagesDir,
-      processRunner: processRunner,
       platform: MockPlatform(),
```

Comment:
You should probably run at least some tests with a `MockPlatform(isWindows: true)` now that we have custom path parsing/construction logic in the tool itself, to make sure it's behaving as expected on Windows.

Reply:
added logic to run all the tests with windows as well as the default.